.github/workflows/publish-swaggerhub.yaml

```diff
@@ -0,0 +1,83 @@
+#
+#  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+#  Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+#
+#  See the NOTICE file(s) distributed with this work for additional
+#  information regarding copyright ownership.
+#
+#  This program and the accompanying materials are made available under the
+#  terms of the Apache License, Version 2.0 which is available at
+#  https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+---
+name: "Publish OpenAPI to Swaggerhub"
+
+on:
+  workflow_call:
+  push:
+    branches:
+      - feature/swagger-api-2
+
+jobs:
+  swagger-api:
+    runs-on: ubuntu-latest
+    env:
+      SWAGGERHUB_API_KEY: ${{ secrets.SWAGGERHUB_API_KEY }}
+      SWAGGERHUB_USER: ${{ secrets.SWAGGERHUB_USER }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup JDK 17
+        uses: actions/setup-java@v3.11.0
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+
+      - name: Install tools
+        run: |
+          npm i -g swaggerhub-cli
+          sudo apt install -y crudini
+
+      - name: Extract versions
+        run: |
+          export DOWNSTREAM_VERSION=$(crudini --get ./gradle.properties '' version)
```

Comment:
I'd prefer having the `DOWNSREAM_VERSION` passed as optional workflow input, and only read it from `gradle.properties` if the input is not present.